File: 02 - Cpp/07 - Diagonal negativos.cpp
#include <iostream>

using namespace std;

int main()
{
	int i, j, n, contador_negativos = 0;

	cout << "Qual a ordem da matriz? ";
	cin >> n;

	int matriz[n][n];

	for (i = 0; i < n; i++)
	{
		for (j = 0; j < n; j++)
		{
			cout << "Elemento [" << i << "," << j << "]: ";
			cin >> matriz[i][j];
		}
	}

	cout << '\n' << "DIAGONAL PRINCIPAL: " << '\n';
	for (i = 0; i < n; i++)
	{
		cout << matriz[i][i] << " ";
	}


	for (i = 0; i < n; i++)
	{
		for (j = 0; j < n; j++)
		{
			if (matriz[i][j] < 0)
			{
				contador_negativos++;
			}
		}
	}
	cout << '\n' << "QUANTIDADE DE NEGATIVOS: " << contador_negativos << '\n';

	return 0;
}

File: 05 - Python/07 - Diagonal negativos.py
n = int(input("Qual a ordem da matriz? "))

matriz: [[int]] = [[0 for x in range(n)] for x in range(n)]

for i in range(0, n):
    for j in range(0, n):
        matriz[i][j] = int(input(f"Elemento [{i},{j}]: "))

print("DIAGONAL PRINCIPAL: ")
for i in range(0, n):
    print(f"{matriz[i][i]} ", end="")
print()

contador_negativos = 0
for i in range(0, n):
    for j in range(0, n):
        if (matriz[i][j] < 0):
            contador_negativos += 1

print(f"QUANTIDADE DE NEGATIVOS: {contador_negativos}")
